content/tutorials/presence.textile

```diff
@@ -0,0 +1,306 @@
+---
+title: Presence Tutorial
+section: tutorials
+index: 10 # TODO ??
+---
+
+Presence enables clients to be aware of other clients that are currently “present” on a channel. Each member present on a channel has a self-assigned name — a "@client ID@" — along with an optional payload that can be used to describe the member’s status or attributes. Presence allows you to quickly build apps such as chat rooms and multiplayer games by automatically keeping track of who is present in real time across any device.
+
+When someone enters or leaves a channel, or updates their member data, a presence event is emitted to all presence subscribers on that channel. Subscribing to presence events makes it incredibly easy to build an app that shows, in real time, any changes to clients connected to Ably and present on a channel.
+
+The optional payload can be a string, JSON object, JSON array, or binary data, same as with message data.
+
+"Using our Presence API":/realtime/presence is easy. Let's get started.
+
+<%= partial 'tutorials/_step-1-setup-free-account' %>
+
+<%= partial 'tutorials/_install_ablyrealtime', vars(use_client_id: true, why_realtime: "be present and subscribe to presence messages") %>
+
+h2. Step 3 - Enter the presence set
+
+With Ably, you can be "attached":/realtime/channels-messages#channels to a channel (that is: be receiving messages published on that channel, or publishing them yourself) without necessarily being present on the channel. If you want to be present, and be seen by other users, you need to explicitly enter the presence set. For example, for a channel called "chatroom":
+
+blang[javascript,nodejs].
+  ```[javascript]
+    var channel = realtime.channels.get('chatroom');
+    channel.attach(function(err) {
+      console.log('If no err, we're now attached to the channel');
```

Comment:
I kind of feel that you should either do `if(!err) { ... }` or get rid of `If no err, ` because it's just confusing in the console seeing `If not err, we're now attached` i.e. are we attached, are we not, what the hell is going on? 😄 

Perhaps instead just a single line above `if (err) { return console.error("Error attaching", err) }`?